src/Schema/Directives/Fields/MiddlewareDirective.php

```diff
@@ -27,52 +63,114 @@ public function name()
      *
      * @return FieldValue
      */
-    public function handleField(FieldValue $value, \Closure $next)
+    public function handleField(FieldValue $value, \Closure $next): FieldValue
     {
-        $checks = $this->getChecks($value);
-
-        if ($checks) {
-            $middlewareRegistry = resolve(MiddlewareRegistry::class);
-
-            if ('Query' === $value->getNodeName()) {
-                $middlewareRegistry->registerQuery(
-                    $value->getFieldName(),
-                    $checks
-                );
-            } elseif ('Mutation' === $value->getNodeName()) {
-                $middlewareRegistry->registerMutation(
-                    $value->getFieldName(),
-                    $checks
-                );
-            }
-        }
+        $middleware = $this->getQualifiedMiddlewareNames(
+            $this->directiveArgValue('checks')
+        );
+        $resolver = $value->getResolver();
 
-        return $next($value);
+        return $next(
+            $value->setResolver(
+                function ($root, array $args, GraphQLContext $context, ResolveInfo $resolveInfo) use ($resolver, $middleware) {
+                    return $this->pipeline
+                        ->send($context->request())
+                        ->through($middleware)
+                        ->then(function (Request $request) use ($resolver, $root, $args, $resolveInfo){
+                            return $resolver(
+                                $root,
+                                $args,
+                                $this->createsContext->generate($request),
```

Comment:
Why is the context regenerated here and not passed through from [line 75](https://github.com/nuwave/lighthouse/pull/395/files#diff-1871bf72c1cbac916932b2c17b024df8R75)?

Reply:
I think this might not be necessary anymore. It might be there because middleware is often used for authentication, so we have to make sure the `Context` is updated with the current user.

If the regeneration is not necessary, i would like to see that covered with a test, then we can remove it. Open to PRs.